### PR TITLE
Make New-ModuleManifest consistent with Update-ModuleManifest

### PR DESCRIPTION
**Pull request by pougetat**
# PR Summary

The purpose of this PR is to make New-ModuleManifest consistent with Update-ModuleManifest by adding `Prerelease`, `RequireLicenseAcceptance`, and `ExternalModuleDependencies` properties.

## PR Context

https://github.com/PowerShell/PowerShell/issues/8585

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/PowerShell/PowerShell/issues/8585
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 1]
```diff
@@ -1197,6 +1217,9 @@ private void BuildPrivateDataInModuleManifest(StringBuilder result, StreamWriter
                 BuildModuleManifest(result, "ProjectUri", Modules.ProjectUri, ProjectUri != null, () => QuoteName(ProjectUri), streamWriter);
                 BuildModuleManifest(result, "IconUri", Modules.IconUri, IconUri != null, () => QuoteName(IconUri), streamWriter);
                 BuildModuleManifest(result, "ReleaseNotes", Modules.ReleaseNotes, !string.IsNullOrEmpty(ReleaseNotes), () => QuoteName(ReleaseNotes), streamWriter);
+                BuildModuleManifest(result, "Prerelease", Modules.Prerelease, !string.IsNullOrEmpty(Prerelease), () => QuoteName(Prerelease), streamWriter);
+                BuildModuleManifest(result, "RequireLicenseAcceptance", Modules.RequireLicenseAcceptance, RequireLicenseAcceptance.IsPresent, () => { return RequireLicenseAcceptance.IsPresent ? "$true" : "False"; }, streamWriter);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]:
```suggestion
                BuildModuleManifest(result, "RequireLicenseAcceptance", Modules.RequireLicenseAcceptance, RequireLicenseAcceptance.IsPresent, () => { return RequireLicenseAcceptance.IsPresent ? "$true" : "$False"; }, streamWriter);
```

**Review by iSazonov — COMMENTED** [review]
Please add tests for new properties.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 2]
```diff
@@ -495,6 +495,26 @@ public object PrivateData
...
+        public string Prerelease { get; set; }
+
+        /// <summary>
+        /// Specify whether the module requires explicit user acceptance for install/update/save.
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter RequireLicenseAcceptance { get; set; }
+
+        /// <summary>
+        /// Specify the ReleaseNotes.
```
[the remark this answers is not in the document]
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Sorry. Copy pasted that. I'll fix soon.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 1 above)
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: I agree with what you are suggesting. Update-ModuleManifest specifies it as "False" and not "$False" which is why I didn't simply go with $false in the first place.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 3]
```diff
@@ -495,6 +495,26 @@ public object PrivateData
         [ValidateNotNullOrEmpty]
         public string ReleaseNotes { get; set; }
 
+        /// <summary>
+        /// Specify that the module is prerelease.
```
[the remark this answers is not in the document]
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Yep. The reason I didn't do that in the first place is that the other parameter comments don't follow this rule but I can update all the comments to follow this standard.

[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: I believe new code shouldn't add new style issues.
Fill free to fix other comments in new PR.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: I don't understand the difference. Could you please point the code or an example?

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Done

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Marking this as resolved then.

**Comment by pougetat** [comment]
I agree that my tests should test for new properties (otherwise they are missing the point of adding new code). Having said that this means I will also need to modify Test-ModuleManifest since this command does not currently return anything for the properties added to New-ModuleManifest in this PR. I will go ahead and do that and then update my tests.

**Comment by pougetat** [comment]
@iSazonov  I have updated the tests :).

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: My request is to use "Gets or sets" for new comments. Optionally fix others in new PR.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 4]
```diff
@@ -495,6 +495,26 @@ public object PrivateData
...
+        /// </summary>
+        [Parameter(Mandatory = false)]
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: No need to set defaults.
```suggestion
        [Parameter]
```
Below too.

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 5]
```diff
@@ -495,6 +495,26 @@ public object PrivateData
...
+        [ValidateNotNullOrEmpty]
+        public string Prerelease { get; set; }
+
+        /// <summary>
+        /// Specify whether the module requires explicit user acceptance for install/update/save.
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter RequireLicenseAcceptance { get; set; }
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: I'd expect that default is true. 
@SteveL-MSFT what do you think?

File: test/powershell/engine/Module/NewModuleManifest.Tests.ps1 [hunk 6]
```diff
@@ -1,13 +1,73 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
+Describe "New-ModuleManifest basic tests" -tags "CI" {
+    BeforeEach {
```
Comment (iSazonov) [inline test/powershell/engine/Module/NewModuleManifest.Tests.ps1]: Why we need create _the same_ directory for each test?
```suggestion
    BeforeAll {
```
The same for next Describe.

File: test/powershell/engine/Module/NewModuleManifest.Tests.ps1 [hunk 7]
```diff
@@ -1,13 +1,73 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
+Describe "New-ModuleManifest basic tests" -tags "CI" {
```
Comment (iSazonov) [inline test/powershell/engine/Module/NewModuleManifest.Tests.ps1]: Please add new line after license.

File: test/powershell/engine/Module/NewModuleManifest.Tests.ps1 [hunk 8]
```diff
@@ -1,13 +1,73 @@
...
+    BeforeEach {
+        $moduleName = 'test'
+        $modulePath = "$TestDrive/Modules/$moduleName"
+        $manifestPath = Join-Path $modulePath "$moduleName.psd1"
+
+        New-Item -Path "$TestDrive/Modules/$moduleName" -ItemType Directory
+    }
+
+    AfterEach {
+        Remove-Item -Recurse -Force -ErrorAction SilentlyContinue $modulePath
```
Comment (iSazonov) [inline test/powershell/engine/Module/NewModuleManifest.Tests.ps1]:
```suggestion
        Remove-Item -Path $modulePath -Recurse -Force -ErrorAction SilentlyContinue
```

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Oh that's my bad. I misunderstood. :)

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 5 above)
Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: In anticipation I set the default to true. :)

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs [hunk 9]
```diff
@@ -1197,6 +1217,9 @@ private void BuildPrivateDataInModuleManifest(StringBuilder result, StreamWriter
                 BuildModuleManifest(result, "ProjectUri", Modules.ProjectUri, ProjectUri != null, () => QuoteName(ProjectUri), streamWriter);
                 BuildModuleManifest(result, "IconUri", Modules.IconUri, IconUri != null, () => QuoteName(IconUri), streamWriter);
                 BuildModuleManifest(result, "ReleaseNotes", Modules.ReleaseNotes, !string.IsNullOrEmpty(ReleaseNotes), () => QuoteName(ReleaseNotes), streamWriter);
+                BuildModuleManifest(result, "Prerelease", Modules.Prerelease, !string.IsNullOrEmpty(Prerelease), () => QuoteName(Prerelease), streamWriter);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Perhaps we could use `nameof()` in `BuildModuleManifest()` like `nameof(Prerelease)`.
It is question only.
/cc  @daxian-dbw

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 9 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: That would be good :)

Comment (iSazonov) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: @pougetat Could you please make the change in all calls of `BuildModuleManifest()`?

Comment (pougetat) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: done

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs (hunk 9 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs]: Fill free to pull new PR or open new tracking issue.

**Review by anmenaga — APPROVED** [review]

**Comment by iSazonov** [comment]
@pougetat Please create new issue in PowerShell-Docs repo and reference in the PR description. (Current link points to source issue that is wrong)

**Pull request merged**